### ISSUE TITLE
Processing files

### ISSUE THREAD

**Issue by nkaramolegos** [jmfriedt/passive_radar#3]
Hello, nice work. Are all the estimation of distance of the target taking place in the goship.m file? I mean is this the overall code for the passive radar?

**Comment by jmfriedt** [comment]
Yes that's all: https://github.com/jmfriedt/passive_radar/blob/main/171210ship/goship.m#L94 computes the cross-correlation between the frequency shifted reference channel https://github.com/jmfriedt/passive_radar/blob/main/171210ship/goship.m#L92 and the surveillance channel, and time is converted to distance when plotting the result https://github.com/jmfriedt/passive_radar/blob/main/171210ship/goship.m#L98 (*3e8 for the speed of light, /fs for the time interval between two samples, /2 for the two way trip, /1000 for km instead of meters since the frequency is in Hz).

**Comment by nkaramolegos** [comment]
Thank you. I am impressed with your overall work in the area. Perfect. Have you measured the range that it can detect targets? For example the maximum distance that it can detect ships  from the coastline?

**Comment by jmfriedt** [comment]
At some point we will end up being limited by the curvature of the Earth ! In this experiment we had the antennas
on the beach at sea level facing the sea, so that a 20 m high ship would vanish under the horizon at a distance of  (acos(6400/(6400+0.02)))*6400=16 km (assuming a 6400-km Earth radius).
At the bottom of the videos of https://github.com/jmfriedt/passive_radar/tree/main/171210ship you will see some dots switching on and off at 8-km range at least so that sounds realistic since grazing angle illumination is probably not the most favorable condition.

**Comment by jmfriedt** [comment]
If I understand your question correctly, this is what we did in http://jmfriedt.free.fr/dvbt_hardware.pdf figure 5. In this article I argue that in order to keep the time delay calibration valid for the whole measurement, the dual SDR receiver system must never stop streaming so I use a 0-MQ publish (UDP like) data transfer from GNU Radio to the Octave processing software, which only records data once the antenna has settled to a stable azimuth and throws the data away otherwise.

**Comment by nkaramolegos** [comment]
Yes, in this case the antenna azimuth was change by hand. I am thinking if it possible to do something like beam-forming/beam-stearing. But as far as the passive radar we are ok.

**Comment by jmfriedt** [comment]
You will need some spatial diversity for beamforming. This was done by moving the ground-based passive radar receiver by W. Feng in http://jmfriedt.free.fr/IGARSS2019.pdf and by using the motion of the spaceborne source in http://jmfriedt.free.fr/SDR-implemented_passive_bistatic_SAR_system_using_Sentinel-1_signal_and_its_experiment_results.pdf

**Comment by nkaramolegos** [comment]
I'm unsure if I fully grasp the concept. My idea, perhaps simplistic, is to employ an array of stationary antennas rather than physically reorienting the receive antenna to target a specific position. The approach involves utilizing beam-forming or beam-steering techniques to direct focus towards the intended target (during RX). Am I overlooking something critical?
